### PR TITLE
Post-purchase Migration Experience: Handle pending and started overview states

### LINKED ISSUE #95383: Post-purchase Migration Experience: Handle selecting a DIFM site in the Sites dashboard that has started migration
## Description
When a user clicks on a site in the Sites dashboard that has the `migration-started-difm` sticker, display a detailed Overview communicating what the user can expect.

Figma - nOWSHCjXMpB5sXHthTVdvK-fi-3949_19811

Image

### PR DESCRIPTION
**Pull request by gabrielcaires**
Closes #95383

## Proposed Changes
* Show different screen  for `migration-pending-*` and `migraiton-started-*` scenarios
* Split the big migration-overview component into smaller and silly ones. 
* Refactor tests to describe better the scenarios.

> [!NOTE]  
> I split the multiple screen variations into small components (MigrationStartedDIY, MigrationStartedDIFM, MigrationPending) to fit better the requirements. I had issues introducing my changes without making them change.  Automated tests fully cover it.

> [!NOTE]  
> The cancel button will be implemented on #96157 

## Testing Instructions
**Scenario 1: Migration Pending DIY**
* Enable the feature flag (DevTools> Console > and run:
sessionStorage.setItem('flags,' 'automated-migration/pending-status')
* Use the `/setup/hosted-site-migration` flow' 
* Follow all steps, choosing the DIY option
* Stop when you arrive at the instructions page
* Get the siteSlug value on the URL
* Open a new browser tab and access `/sites` 
* Find and click on the created site (using the siteSlug that you got previously) to open the overview page. Please pay attention you need to remove the `.wordpress.com` domain to find your site
* Check if you can see the screen 1

**Scenario 2: Migration Pending DIFM**
*  Run another migration, following the item described in scenario 1, but choose the DIY option and stop on the instructions page.
* Get the siteSlug value on the URL
* Open a new browser tab and access `/sites` 
* Find and click on the created site (using the siteSlug that you got previously) to open the overview page. Please pay attention. You need to remove the `.wordpress.com` domain to find your site
* Check if you can see the screen 2

**Scenario 3: Migration Started DIY**
* Run another migration, following the item described in scenario 1, but choose the DIY option.
* On the instructions page, follow  instructions  and really start a migration
*  Open a new browser tab and access `/sites` 
*  Select your site and compare it with the screenshot 4

> [!NOTE]  
> The migration started status is set via backend based on the real migration state, so it may have a time to see the status reflected on the /sites

**Scenario 4: Migration Started DIFM**
* Run another migration, following the item described in scenario 1, but choose the DIY option.
* On the credentials step, set the correct credentials and follow the flow until the last step
*  Open a new browser tab and access `/sites` 
*  Select your site and compare it with the screenshot 3

## Screenshots

1. Pending - DIY
image

2. Pending - DIFM
image

3. Started - DIY
image

4. Started - DIFM
image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/hosting/overview/components/test/migration-overview.tsx [hunk 1]
```diff
@@ -1,115 +0,0 @@
-/**
```
Comment (gabrielcaires) [inline client/hosting/overview/components/test/migration-overview.tsx:1]: The tests are now on `client/hosting/overview/components/migration-overview/test/index.tsx`

File: client/hosting/overview/components/migration-overview/index.tsx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+		// For the flows where the checkout is after the choice.
+		switch ( migrationType ) {
+			case 'diy':
+				return addQueryArgs(
+					{
+						...baseQueryArgs,
+						destination: 'upgrade',
+						how: 'myself',
+					},
+					'/setup/site-migration/site-migration-upgrade-plan'
```
Comment (gabrielcaires) [inline client/hosting/overview/components/migration-overview/index.tsx]: It doesn't make more sense

**Comment by andres-blanco** [comment]
Content is not horizontally centered
image

File: client/hosting/overview/components/migration-overview/style.scss [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		}
+
+		.migration-started-difm__item {
+			display: flex;
+			gap: .75rem;
+			font-size: 1rem;
+			align-items: center;
+		}
+		.migration-started-difm__icon {
+				fill: #3858E9;
```
Comment (andres-blanco) [inline client/hosting/overview/components/migration-overview/style.scss]: formatting is weird here

File: client/hosting/overview/components/migration-overview/style.scss [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		}
+		.migration-started-difm__icon-wrapper {
+			width: 52px;
+			height: 52px;
+			// box-sizing: border-box;
```
Comment (andres-blanco) [inline client/hosting/overview/components/migration-overview/style.scss]: should we keep this commented code?

File: client/hosting/overview/components/migration-overview/style.scss [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+			display: flex;
+			align-items: center;
+			justify-content: center;
+			background-color: #F7F8FE;
+			// padding: 1rem;
```
Comment (andres-blanco) [inline client/hosting/overview/components/migration-overview/style.scss]: should we keep this commented code?

File: client/hosting/overview/components/migration-overview/style.scss (hunk 4 above)
Comment (gabrielcaires) [reply client/hosting/overview/components/migration-overview/style.scss]: Fixed here 717f480ecd2326edf7626ff0b0999d3a56d0f8bd

File: client/hosting/overview/components/migration-overview/style.scss (hunk 3 above)
Comment (gabrielcaires) [reply client/hosting/overview/components/migration-overview/style.scss]: Fixed [717f480](https://github.com/Automattic/wp-calypso/pull/95963/commits/717f480ecd2326edf7626ff0b0999d3a56d0f8bd)

File: client/hosting/overview/components/migration-overview/style.scss (hunk 5 above)
Comment (gabrielcaires) [reply client/hosting/overview/components/migration-overview/style.scss]: Fixed [dbc14a9](https://github.com/Automattic/wp-calypso/pull/95963/commits/dbc14a94f4d8fea4f309aac4d0bbc3323247a115)

File: client/hosting/overview/components/migration-overview/style.scss (hunk 4 above)
Comment (gabrielcaires) [reply client/hosting/overview/components/migration-overview/style.scss]: [dbc14a9](https://github.com/Automattic/wp-calypso/pull/95963/commits/dbc14a94f4d8fea4f309aac4d0bbc3323247a115)

**Comment by andres-blanco** [comment]
> Linking to the DIY step has an issue, because we are not adding the source site in the link, not sure still what the solution is but I wanted to get it written
> 
> image

more discussion here p1731007302987479/1730920947.410109-slack-C07N53UNUQJ

**Comment by gabrielcaires** [comment]
> Linking to the DIY step has an issue, because we are not adding the source site in the link, not sure still what the solution is but I wanted to get it written
> 

I also don't like it, but we probably need more exploration to solve this because the UI doesn't have information about the source site.

**Comment by gabrielcaires** [comment]
> Content is not horizontally centered on big screens

All other tab contents are not centered:
image

**Comment by andres-blanco** [comment]
> All other tab contents are not centered:

That's not good, I'll make a issue to tackle this

**Comment by andres-blanco** [comment]
> Linking to the DIY step has an issue, because we are not adding the source site in the link, not sure still what the solution is but I wanted to get it written

The instructions screen adjusts the copy to not provide broken links if the source URL is not present. While it's not great, it's not broken either. More details here: p1731328427817849/1730920947.410109-slack-C07N53UNUQJ

**Review by andres-blanco — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16999883

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @gabrielcaires for including a screenshot in the description! This is really helpful for our translators.

**Pull request merged**